app/controllers/incomeSources/add/IncomeSourcesAccountingMethodController.scala

```diff
@@ -0,0 +1,255 @@
+          btaNavPartial = user.btaNavPartial
+        )(user, messages)))
+    }
+  }
+
+  private def loadIncomeSourceAccountingMethod(isAgent: Boolean, errorHandler: ShowInternalServerError, incomeSourceType: String)
+                                              (implicit user: MtdItUser[_], backUrl: String, postAction: Call, messages: Messages): Future[Result] = {
+    Future.successful(Ok(view(
+      incomeSourcesType = incomeSourceType,
+      form = IncomeSourcesAccountingMethodForm(incomeSourceType),
+      postAction = postAction,
+      isAgent = isAgent,
+      backUrl = backUrl,
+      btaNavPartial = user.btaNavPartial
+    )(user, messages)))
+  }
+
+  def handleRequest(isAgent: Boolean, incomeSourceType: String)
+                   (implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext, messages: Messages): Future[Result] = {
+
+    val incomeSourcesEnabled: Boolean = isEnabled(IncomeSources)
+    val backUrl = incomeSourceType match {
+      case SelfEmployment.key =>
+        if (isAgent)
+            routes.AddBusinessAddressController.showAgent().url
+        else
+            routes.AddBusinessAddressController.show().url
+      case UkProperty.key =>
+        if (isAgent)
+            routes.AddIncomeSourceStartDateCheckController.showUKPropertyAgent.url
+        else
+            routes.AddIncomeSourceStartDateCheckController.showUKProperty.url
+      case ForeignProperty.key =>
+        if (isAgent)
+            routes.AddIncomeSourceStartDateCheckController.showForeignPropertyAgent.url
+        else
+            routes.AddIncomeSourceStartDateCheckController.showForeignProperty.url
+    }
+    val postAction: Call = if (isAgent) controllers.incomeSources.add.routes.IncomeSourcesAccountingMethodController.submitAgent(incomeSourceType) else
+      controllers.incomeSources.add.routes.IncomeSourcesAccountingMethodController.submit(incomeSourceType)
+
+    val errorHandler: ShowInternalServerError = if (isAgent) itvcErrorHandlerAgent else itvcErrorHandler
+
+    if (incomeSourcesEnabled) {
+      if(incomeSourceType == SelfEmployment.key) {
+        handleUserActiveBusinessesCashOrAccruals(isAgent = isAgent, errorHandler = errorHandler, incomeSourceType)(
+          user = user, backUrl = backUrl, postAction = postAction, messages = messages)
+      } else {
+        loadIncomeSourceAccountingMethod(isAgent = isAgent, errorHandler = errorHandler, incomeSourceType)(
+          user = user, backUrl = backUrl, postAction = postAction, messages = messages)
+      }
+    } else {
+      Future.successful(Ok(customNotFoundErrorView()(user, messages)))
+    } recover {
+      case ex: Exception =>
+        Logger("application").error(s"${if (isAgent) "[Agent]"}" +
+          s"Error getting BusinessEndDate page: ${ex.getMessage}")
+        errorHandler.showInternalServerError()
+    }
+  }
+
+   private def actionIndividualSubmitRequest(incomeSourceType: String): (Call, String, Call) = {
+    incomeSourceType match {
```

Comment:
you still pattern matching over value you passing as a parameter
why ?
(routes.IncomeSourcesAccountingMethodController.submit(incomeSourceType)

Reply:
Missed updating the value param, I updated it now.